### PR TITLE
Add node labels to Jenkins Pipelines traces

### PR DESCRIPTION
**Pull request by drodriguezhdez**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

This PR adds the Node Labels to the Jenkins Pipelines traces using the tag `ci.node.labels`.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java [hunk 1]
```diff
@@ -771,4 +790,31 @@ public static String toISO8601(Date date) {
...
+     * @return json array string
+     */
+    public static String toJson(final Set<String> set) {
+        if(set == null || set.isEmpty()) {
+            return "";
+        }
+
+        // We want to avoid using Json libraries cause
+        // may cause incompatibilities on different Jenkins versions.
+        final StringBuilder sb = new StringBuilder();
```
Comment (AdrianLC) [inline src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java:805]: can there be quotes in Jenkins labels?

File: src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildPipeline.java [hunk 2]
```diff
@@ -133,6 +133,7 @@ private void completeInformation(final List<BuildPipelineNode> nodes, final Buil
                 final BuildPipelineNode executableChildNode = searchExecutableChildNode(node);
                 if(executableChildNode != null) {
                     node.setPropagatedNodeName(executableChildNode.getNodeName());
+                    node.setPropagatedNodeLabels(executableChildNode.getNodeLabels());
```
Comment (AdrianLC) [inline src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildPipeline.java:136]: are you propagating it from job spans to stage/pipeline? I guess that's accurate for Jenkins?

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java (hunk 1 above)
Comment (drodriguezhdez) [reply src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java:805]: If you mean double quotes: no, Jenkins removes the double quotes (`"`) when you configure the labels for the worker.
Other characters are allowed (`'`, `$`, `%`, etc..)

File: src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildPipeline.java (hunk 2 above)
Comment (drodriguezhdez) [reply src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildPipeline.java:136]: Yes, the worker information only is available in an actual executable child node, which is a `job`.

**Review by AdrianLC — DISMISSED** [review]

**Review by hithwen — APPROVED** [review]

**Pull request merged**